### DISCUSSION TITLE
GPSR API responsible producers and persons information?

### DISCUSSION THREAD

**Discussion by MBI910** [allegro/allegro-api#10299] [Suggestions - resolved]
Hi there, 

I am looking for all the API fields that are available in the backend of the allegro website:
https://salescenter.allegro.com/responsible-producers
![image](https://github.com/user-attachments/assets/3df30323-561e-4bf1-a8d7-4bf6edd01a9d)

https://salescenter.allegro.com/responsible-persons 
![image](https://github.com/user-attachments/assets/d4f57244-553c-4527-aa5d-ea8c56ee4957)

Until now, I can not find a fully comprehensive documentation where I can add all these field to our products

This was not really helpful https://github.com/allegro/allegro-api/issues/9958

**Comment by AureliuszBrussy** [comment]
Hi,

you can: 
- Create responsible person: [POST /sale/responsible-persons](https://developer.allegro.pl/documentation#operation/responsiblePersonsPOST)
- Create responsible producer: [POST /sale/responsible-producers](https://developer.allegro.pl/documentation#operation/responsibleProducersGET). 

You can also check and edit created entities. Read more about all available endpoints [in our guide](https://developer.allegro.pl/tutorials/account-and-user-data-management-jn9vBjqjnsw#responsible-producers-contact-information).

You can add a responsible person and producer to the offer using **[/sale/product-offers](https://developer.allegro.pl/documentation#operation/createProductOffers)** resources. 

Save this data in one of the objects:
- `productSet.product.responsiblePerson`
- `productSet.product.responsibleProducer`.

**Reply by MBI910** [reply]
thank you!

do you know when there will be the possibility to use text for https://developer.allegro.pl/news/gpsr-we-allow-you-to-add-product-safety-information-D7j2PRy6eho ?

![image](https://github.com/user-attachments/assets/2bcd32c9-c5c2-4d21-881d-2e02c1cc9474)

**Reply by AureliuszBrussy** [reply]
Unfortunately, I am not able to provide a specific date yet. We will, of course, communicate this, so follow [our news](https://developer.allegro.pl/news) or [newsletter](https://developer.allegro.pl/newsletter).
